### PR TITLE
propose relative url for pagination next link

### LINKED ISSUE #93: Manifest Referrers API: Pagination: Allow relative url for next link
> https://github.com/oras-project/artifacts-spec/blob/v1.0.0-draft.1/manifest-referrers-api.md#paging-results 

> If there are more items, the URL for the next collection is encoded in a [RFC5988](https://datatracker.ietf.org/doc/html/rfc5988) Link header, as a "next" relation. Clients SHOULD treat this as an opaque value and not try to construct it themselves.

In current v1.0.0-draft.1 spec, the next link in pagination result is an absolute url that the client will treat as an opaque value. The challenge of the requirement is the absolute url will not work if the registry is running behind a gateway/proxy. The following flow illustrates the problem. In this case, client is only able to connect to `myregistry.gateway.io`.

client => myregistry.gateway.io(gateway) => myregistry.azurecr.io (registry)

If the next link returned by the registry is an absolute path, eg, `http://myregistry.azurecr.io/oras/artifacts/v1/hello-world/manifests/sha256:3c3a4604a545cdc127456d94e421cd355bca5b528f4a9c1905b15da2eb4a4c6b/referrers?n=5&nextToken=abc`, the client won't be able to follow the link to request the next page items.

We propose to loose the constraint to allow relative path in the paging result. For the above example, the next link will be `/oras/artifacts/v1/hello-world/manifests/sha256:3c3a4604a545cdc127456d94e421cd355bca5b528f4a9c1905b15da2eb4a4c6b/referrers?n=5&nextToken=abc`. 
The client will parse the url and compose the full url with the appropriate host: `https://myregistry.gateway.io/oras/artifacts/v1/hello-world/manifests/sha256:3c3a4604a545cdc127456d94e421cd355bca5b528f4a9c1905b15da2eb4a4c6b/referrers?n=5&nextToken=abc`

### PR DESCRIPTION
**Pull request by northtyphoon**
resolve #93 

Signed-off-by: Bin Du <bindu@microsoft.com>

### CODE REVIEW

**Review by SteveLasker — DISMISSED** [review]
LGTM,
Thanks, @northtyphoon, this is a great example of iterating the spec as its being implemented, identifying gaps enabling real-world implementations.

File: manifest-referrers-api.md [hunk 1]
```diff
@@ -182,10 +182,17 @@ http://example.com/v2/hello-world/_oras/artifacts/referrers?digest=sha256:3c3a46
...
+
 ```
 <http://example.com/v2/hello-world/_oras/artifacts/referrers?digest=sha256:3c3a4604a545cdc127456d94e421cd355bca5b528f4a9c1905b15da2eb4a4c6b&n=5&nextToken=abc>; rel="next"`.
 ```
 
+or 
+
+```
+</_oras/artifacts/referrers?digest=sha256:3c3a4604a545cdc127456d94e421cd355bca5b528f4a9c1905b15da2eb4a4c6b&n=5&nextToken=abc>; rel="next"`.
+```
```
Comment (sajayantony) [inline manifest-referrers-api.md:187]: We need to fix this example to - 

`/v2/hello-world/_oras/artifacts/referrers?digest=sha256:3c3a4604a545cdc127456d94e421cd355bca5b528f4a9c1905b15da2eb4a4c6b&n=5&nextToken=abc>; rel="next"`

Missing the /v2/{repo} base path

**Review by sajayantony — CHANGES_REQUESTED** [review]

**Comment by sajayantony** [comment]
@michaelb990 should we recommend only relative URLs?

**Comment by SteveLasker** [comment]
@northtyphoon, can you fix your DCO issue on the "update example" commit?

**Comment by michaelb990** [comment]
> @michaelb990 should we recommend only relative URLs?

I would prefer to just have 1 way to do things where possible. Seems like the full URL doesn't work for some clients, and I don't see any downsides to just using the relative link. What do you think?

**Comment by SteveLasker** [comment]
Fewer choices make things simpler. (👍)  @northtyphoon, can you update?

**Comment by sajayantony** [comment]
+1 on making relative the only option.

**Review by SteveLasker — APPROVED** [review]
LGTM

**Review by sajayantony — APPROVED** [review]

**Review by michaelb990 — APPROVED** [review]

**Pull request merged**